awscli/customizations/configure/mfalogin.py

```diff
@@ -0,0 +1,327 @@
+        except ClientError as e:
+            sys.stderr.write(f"An error occurred: {e}\n")
+            return None
+
+    def _setup_session_with_profile(self, source_profile):
+        """Setup and validate session with the given profile."""
+        try:
+            session = botocore.session.Session(profile=source_profile)
+            
+            if (source_profile not in session.available_profiles 
+                and source_profile != 'default'):
+                sys.stderr.write(f"The profile ({source_profile}) could not be found. \n")
+                return None, None
+            
+            credentials = session.get_credentials()
+            if credentials is None:
+                if source_profile == 'default':
+                    return None, None  # Signal to handle missing default profile
+                else:
+                    sys.stderr.write(f"Unable to locate credentials for profile {source_profile}\n")
+                    return None, None
+            
+            return session, session.get_scoped_config()
+            
+        except ProfileNotFound:
+            if source_profile == 'default':
+                return None, None  # Signal to handle missing default profile
+            else:
+                sys.stderr.write(f"The profile ({source_profile}) could not be found. \n")
+                return None, None
+        except Exception as e:
+            sys.stderr.write(f"Error accessing profile {source_profile}: {str(e)}\n")
+            return None, None
+
+    def _resolve_mfa_serial(self, parsed_args, source_config):
+        """Resolve MFA serial from args, config, or prompt."""
+        mfa_serial = parsed_args.serial_number or source_config.get('mfa_serial')
+        if not mfa_serial:
+            mfa_serial = self._prompter.get_credential_value(
+                'None', 'mfa_serial', 'MFA serial number or ARN'
+            )
+            if not mfa_serial:
+                sys.stderr.write("MFA serial number or MFA device ARN is required\n")
+                return None
+        return mfa_serial
+
+    def _write_temporary_credentials(self, temp_credentials, target_profile):
+        """Write temporary credentials to the credentials file."""
+        credentials_file = os.path.expanduser(self._session.get_config_variable('credentials_file'))
+
+        credential_values = {
+            '__section__': target_profile,
+            'aws_access_key_id': temp_credentials['AccessKeyId'],
+            'aws_secret_access_key': temp_credentials['SecretAccessKey'],
+            'aws_session_token': temp_credentials['SessionToken'],
+        }
+
+        try:
+            expiration_time = temp_credentials['Expiration'].strftime(
+                '%Y-%m-%d %H:%M:%S UTC'
+            )
+        except AttributeError:
+            expiration_time = str(temp_credentials['Expiration'])
+
+        credential_values['#Credentials expire at: '] = (
+            f"{expiration_time}"
+        )
```

Comment:
It might be happening in `writer.py`, but as I refresh by re-running `mfa-login` I'm seeing the expiration lines stack up
```
[12345-google-authenticator]
aws_access_key_id = ...
aws_secret_access_key = ...
aws_session_token = ...
#Credentials expire at:  = 2025-08-28 01:09:31 UTC
#Credentials expire at:  = 2025-08-28 01:50:53 UTC
#Credentials expire at:  = 2025-08-28 01:51:16 UTC
#Credentials expire at:  = 2025-08-28 01:51:38 UTC
```

Reply:
I'm not in love with the `:  = ` syntax, but not sure it's worth officially plumbing through "comment support" in this PR. Agree it's helpful to stamp this for those reading the file.

An upcoming project might need an actual expiration field, so if that shakes out we can convert this.